backend/service/topology/querybuilder.go

```diff
@@ -0,0 +1,115 @@
+package topology
+
+import (
+	"fmt"
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+
+	topologyv1 "github.com/lyft/clutch/backend/api/topology/v1"
+)
+
+const (
+	column   = "column."
+	metadata = "metadata."
+)
+
+func paginatedQueryBuilder(
+	filter *topologyv1.SearchTopologyRequest_Filter,
+	sort *topologyv1.SearchTopologyRequest_Sort,
+	pageToken,
+	limit int,
+) (sq.SelectBuilder, error) {
+	queryLimit := 100
+	if limit > 0 {
+		queryLimit = limit
+	}
+
+	queryOffset := 0
+	if pageToken > 0 {
+		queryOffset = pageToken * limit
+	}
+
+	query := sq.StatementBuilder.PlaceholderFormat(sq.Dollar).
+		Select("id, data, metadata").
+		From("topology_cache").
+		Limit(uint64(queryLimit)).
+		Offset(uint64(queryOffset))
+
+	query = filterQueryBuilder(query, filter)
+	query = sortQueryBuilder(query, sort)
+
+	return query, nil
+}
+
+func filterQueryBuilder(query sq.SelectBuilder, f *topologyv1.SearchTopologyRequest_Filter) sq.SelectBuilder {
+	if f.Search != nil && len(f.Search.Field) > 0 {
+		if strings.HasPrefix(f.Search.Field, column) {
+			query = query.Where(sq.Like{strings.TrimPrefix(f.Search.Field, column): fmt.Sprintf("%%%s%%", f.Search.Text)})
+		} else if strings.HasPrefix(f.Search.Field, metadata) {
+			mdQuery := convertMetadataToQuery(strings.TrimPrefix(f.Search.Field, metadata))
+			query = query.Where(sq.Like{mdQuery: fmt.Sprintf("%%%s%%", f.Search.Text)})
```

Comment:
What do you think about using [ilike](https://sourcegraph.com/github.com/Masterminds/squirrel/-/blob/expr.go#L272) for insensitive searching?

Reply:
Also, I think you got an extra percent in `%%%s%%`

Reply:
I could an option to toggle case sensitive search, so the caller can specify the exact behavior. Since that requires a proto change I can do that in a follow up PR.

No extra precent here, just escaping for fmt.
https://golang.org/pkg/fmt/#hdr-Printing
> %%	a literal percent sign; consumes no value

Reply:
Probably not necessary to support it immediately.

Yeah, my mistake with the %%%

Reply:
Ill leave a todo and follow up.

Reply:
fmt.Sprint is a little cleaner and easier to read maybe, `fmt.Sprint("%", f.Search.Text, "%"). doesn't really matter if there are tests though.